crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -544,4 +546,26 @@ mod tests {
         // SAFETY: drop is able to drop a value of its `item_layout`
         let _ = unsafe { BlobVec::new(item_layout, Some(drop), 0) };
     }
+
+    #[test]
+    fn aligned_zst() {
```

Comment:
None of this actually tests the alignment of the ZST. Is that even possible?

A note should be made here that this is explicitly for UB checking with miri.